airflow/models/baseoperator.py

```diff
@@ -386,8 +386,8 @@ def __init__(
         self.retries = retries
         self.queue = queue
         self.pool = pool
-        self.pool_slots = pool_slots
-        if self.pool_slots < 1:
+        self.pool_slots = 0 if self.__class__.__name__ == 'DummyOperator' else pool_slots
+        if self.pool_slots < 0 or (self.pool_slots == 0 and self.__class__.__name__ != 'DummyOperator'):
```

Comment:
Will this work with ExternalTaskMarker?

Reply:
how is ExternalTaskMarker relevant? admittedly i've never heard of it!

Reply:
```
class ExternalTaskMarker(DummyOperator):
```

Reply:
what do u suggest? I think its fine for ExternalTaskMarker not to consume a slot

Reply:
Does this change make sense at all if my PR has been merged?  It seems to me that the condition will never be met.
https://github.com/apache/airflow/pull/7880